SubmissionTest.java

```diff
@@ -148,15 +148,21 @@ public void testZip() throws IOException
         
         // Get actual Submission filenames.
         File actualDir = new File("zips");
-        File[] actualZips = actualDir.listFiles();
-        Arrays.sort(actualZips);
-        String[] actual = new String[actualZips.length];
-        for (int i = 0; i < actual.length; i++)
+        
+        // Make sure directory has not been deleted 
+        // by deleteZips test or application. 
+        if (actualDir.exists())
         {
-            actual[i] = actualZips[i].getName();
-        }
+            File[] actualZips = actualDir.listFiles();
+            Arrays.sort(actualZips);
+            String[] actual = new String[actualZips.length];
+            for (int i = 0; i < actual.length; i++)
+            {
+                actual[i] = actualZips[i].getName();
+            }
 
-        assertArrayEquals("should be same", expected, actual);
+            assertArrayEquals("should be same", expected, actual);
```

Comment:
@KaraBeason I was worried about that as well, but I don't see a way to make these tests isolated and still functional, given that we can't control the order the tests are run? Or we just deal with the occasional failure of these tests when deleteZips has already been called? Or do we remove the deleteZips method altogether? What do you think is the best course of action?